src/App.vue

```diff
@@ -37,6 +33,28 @@ export default {
       loading: Loading,
     }),
   },
+  mounted() {
+    // dynamically insert map routes
+    mapData.init(baseUrl).then(() => {
+      const mapNames = mapData.getMapNames();
+      mapNames.forEach((mapName) => {
+        this.$router.options.routes.push({
+          name: mapName,
+          path: `/${mapName}`,
+          component: () => import(/* webpackChunkName: "map" */ "./Map.vue"),
+          props: {
+            mapData,
+            postScriptum,
+            initialMap: mapName,
+          },
+        });
+        this.$router.options.routes.push({
+          path: "/", redirect: mapNames[0],
+        });
+      });
+      this.$router.addRoutes(this.$router.options.routes);
+    });
```

Comment:
Haven't I done this with:

```javascript
this.$router.options.routes.push({
  path: "/", redirect: mapNames[0],
});
```

And I mean we can be sure that there will be at least one map available, right? (Since it does not depend on one particular map being available)

Reply:
Have you tried adding a last route like this:
```javascript
// fallback to first map, if name in route could not be matched
this.$router.options.routes.push({
  path: "/*", redirect: mapNames[0],
});
```
Might do the trick.